### ISSUE TITLE
"List Index out of range"

### ISSUE THREAD

**Issue by chrisale** [steepleian/weewx-Weather34#18]
When I run 
sudo python w34_installer.py
and select the macOS.conf file (3), I am getting back a 
"list index out of range"
message. This appears to be a python related issue?

The installation seems to copy files successfully, but I haven't yet completed the installation.

Note that I modified the macOS.conf file in order to match my configuration.  I've attached my macOS.conf file (with a txt extension to satisfy Github) for reference in case it helps.

[macos.conf.txt](https://github.com/steepleian/weewx-Weather34/files/4541145/macos.conf.txt)

**Comment by steepleian** [comment]
Thanks Chris. You won’t hear from me for an hour or so. My other big hobby,
cooking dinner is taking centre stage, 18:30 here now back around 22:00 my
time.

**Comment by chrisale** [comment]
No problem Ian!
This may be related to this same issue so I'll put it here:
The installation guide says:
"The installer will prompt you for the location of your WeeWX webserver folder. The default is set to /var/www/html/weewx. If your path is the same, just press enter otherwise you must type in your full path."
I have tried both macos.conf and setup_py.conf and neither have prompted me for a path.
I set the path by manually editing the conf file, which, I am not sure yet is going to be entirely successful.

**Comment by chrisale** [comment]
This error may also be preventing modifications to the weewx.conf file from occurring as I removed the mentions of weather34 in my weewx.conf before trying this new install and nothing returned to weewx.conf when I ran the installer.

**Comment by chrisale** [comment]
After fiddling around with this I think it's safe to say that the w34installer.py is not completing.  The permissions are not being set, the weewx.conf file is not being modified, and the setup .conf file settings are not being being honoured completely.

Even after settings the conf files with the proper paths for my environment to /Users/Shared manually, the installer sent some skins to a brand new /etc/weewx/skins directory that never existed before.

The w34Highcharts-day also doesn't appear to have a location in weewx.conf ? And so the json_day directory and files are not being generated.

**Comment by steepleian** [comment]
Hi Chris,
Have have made a change to the macos install configuration which should
correct things. You, as far as I know, are the only macOS user that is
using Weather34. Your permissions should be set correctly for your local
install but you may need to set these manually for your remote location.
Thanks,
Ian

**Comment by chrisale** [comment]
Hi Ian,

Thanks for that.  Might want to check the previous two comments re: 

""The installer will prompt you for the location of your WeeWX webserver folder. The default is set to /var/www/html/weewx. If your path is the same, just press enter otherwise you must type in your full path.""

And the modifications to the weewx.conf file which aren't completing.

Thanks for your work!

**Comment by steepleian** [comment]
Sorry I missed your attachment. Paths are now updated in macos.conf and install guide also updated. I did miss one error in your conf file. The w34highcharts folder should sit in weather34 folder rather than public_html.

**Comment by chrisale** [comment]
Thank you very much.

Any idea on the "list index out of range" message and lack of prompt for web directory?

Also, for the w34highcharts, the HighCharts Guide mentions that a w34highcharts-day folder is created for a json_data folder. Is this correct? Also have questions about the config.js file.

**Comment by steepleian** [comment]
Chris,
When you say lack of prompt, was that after the update to the install
config?

**Comment by chrisale** [comment]
Yes, it occurred before and after the changes and with both macos.conf and setup_py.conf.  The installer only copied files, it did not complete the permission changes nor the changes to weewx.conf.

**Comment by steepleian** [comment]
Using the setup_py option puts things in the wrong place for a Mac. I just
ran the macOS option on my Mac. Although I don’t have WeeWX on it, it put
everything in the right place. It’s very puzzling as to what is happening
with yours.

Have you tried doing a manual install?

**Comment by chrisale** [comment]
Yes, I just did the setup version once to see if the same message appeared, which it did. 
Copying the files doesn’t seem to be the issue, all files are copied correctly. But permissions are not set and weewx.conf is not modified. I had to do it all manually in order to get it working.

**Comment by steepleian** [comment]
Ok I think we have to look at the Mac option more closely. I will install
WeeWX on my Mac and take it from there. Are things working ok with your
manual intervention?

**Comment by chrisale** [comment]
Very close! I have checked and rechecked the permissions and locations for the highchairs json files but while the long term charts are working, clicking on a value in the yearly charts brings up the dreaded 
"NO DATA FOUND. Please choose another date. temp_week.json"

json files in 
weather34/w34highcharts/json 
are updating but files in 
weather34/w34highcharts/json_day

are not.

**Comment by steepleian** [comment]
How much data do you have and what are your permissions set to.

**Comment by chrisale** [comment]
The station has data going back to 2006.

Everything in weather34 is 755 on both the local and remote side except wee_reports_w34 which is 777. I am noticing now that when the files are generated, archive_data.php and the files in the json directory are being generated with 644 permissions.

Files in w34highcharts/json_day are 755.

Cheers
Chris

**Comment by steepleian** [comment]
Ok Chris have figured out the installer problem. When I changed the path to
the web server in macOS.conf it broke w34_installer.py. Now got to work out
a solution.

**Comment by steepleian** [comment]
Chris,
There is a new update to the installer. It does select the correct config now when option 3 is chosen. The list index is still out of range however so further tweaks to be made. In the meantime do you have www-data set up as a user and a group. If not this may be a cause of you not being able to create the json_day files.

**Comment by chrisale** [comment]
Are the json_day files created by the user on the web end, or by weewx? (or both?)

**Comment by chrisale** [comment]
And to answer your question, my web host has it set up so that the apache and nginx users are part of the same group and that group co-owns my web directories along with my personal user account.  As I type this, I think I will try setting permissions to 775 rather than 755 on the web end.

**Comment by chrisale** [comment]
*nope: 775, even 777 on everything didn't get the day charts going. I'm thinking I must have them in the wrong directories maybe? Could you send me a screenshot of your correct directory structure for the weather34 and highcharts directories?

**Comment by steepleian** [comment]
Screen shots attached. The first three are my setup on my Pi installed
using setup_py method

The 4th is on my Mac installed using the macos method.

**Comment by chrisale** [comment]
Sorry Ian, I think you forgot to attach the screenshots? :)

**Comment by steepleian** [comment]
No they were attached. Maybe one of our email providers chocked on them. I
will try one at a time. They will be with you tomorrow morning.

**Comment by chrisale** [comment]
Hi Ian,
Sorry, I still haven't received any attachments.  Perhaps responding through the GitHub site and attaching the screenshots directly in the thread here would work better. Thanks for your patience.

Cheers
Chris

**Comment by steepleian** [comment]
Hi Chris,

Sorry for the late reply. Ye Olde Ancient MAC’s hard drive has died. It
served me well for many years. RIP.

Will try and get them to you later when I have recovered the images from
iCloud.

Cheers,
Ian

**Comment by steepleian** [comment]
Ok Chris,

Here they come one-by-one

**Comment by steepleian** [comment]
At least iCloud has not let me down!

**Comment by chrisale** [comment]
Hehe. Hoping for the best! If they still don’t come through, you can always send them on iMessage. My same email applies: chrisale at gmail.com :)

**Comment by balloonhotair** [comment]
Hi,
I am in the process of testing the changes to be able to rsync day_plot data from the weewx machine to the webserver machine and have a question for Chris. I understand that currently you are using rsync to move the json chart data from the weewx machine to the webserver machine. Are you using a cron job outside of weewx? If so how are your dealing with the password? Is it in the crontab entry or have you configured passwordless ssh usinf public/private keys?
Thanks Jerry

**Comment by chrisale** [comment]
Hi Jerry,

Yes I am using keys for rsync.

Cheers
Chris
